BungeeCord-Patches/0064-Adventure.patch

```diff
@@ -0,0 +1,1668 @@
+ 
+     /**
+      * Clears the header and footer displayed in the tab player list.
++     *
++     * @deprecated in favour of using empty components with {@link net.kyori.adventure.audience.Audience#sendPlayerListHeaderAndFooter(net.kyori.adventure.text.Component, net.kyori.adventure.text.Component)}
+      */
++    @Deprecated // Waterfall - we're going on an adventure!
+     void resetTabHeader();
+ 
+     /**
+@@ -338,7 +384,9 @@ public interface ProxiedPlayer extends Connection, CommandSender
+      *
+      * @param title The title to send to the player.
+      * @see Title
++     * @deprecated in favour of {@link net.kyori.adventure.audience.Audience#showTitle(net.kyori.adventure.title.Title)}
+      */
++    @Deprecated // Waterfall - we're going on an adventure!
+     void sendTitle(Title title);
+ 
+     /**
+@@ -387,4 +435,11 @@ public interface ProxiedPlayer extends Connection, CommandSender
+      * @return this player's {@link Scoreboard}
+      */
+     Scoreboard getScoreboard();
++
++    // Waterfall start - we're going on an adventure!
++    @Override
++    default net.kyori.adventure.identity.@org.checkerframework.checker.nullness.qual.NonNull Identity identity() {
++        return net.kyori.adventure.identity.Identity.identity(getUniqueId());
++    }
++    // Waterfall end
+ }
+diff --git a/api/src/main/java/net/md_5/bungee/api/event/LoginEvent.java b/api/src/main/java/net/md_5/bungee/api/event/LoginEvent.java
+index a13eba79..d431fcdd 100644
+--- a/api/src/main/java/net/md_5/bungee/api/event/LoginEvent.java
++++ b/api/src/main/java/net/md_5/bungee/api/event/LoginEvent.java
+@@ -29,7 +29,10 @@ public class LoginEvent extends AsyncEvent<LoginEvent> implements Cancellable
+      * Message to use when kicking if this event is canceled.
+      */
+     @Setter(AccessLevel.NONE)
+-    private BaseComponent[] cancelReasonComponents;
++    // Waterfall start - we're going on an adventure!
++//    private BaseComponent[] cancelReasonComponents;
++    private net.kyori.adventure.text.Component cancelReason = net.kyori.adventure.text.Component.empty();
++    // Waterfall end
+ 
+     // Waterfall start - adding the LoginResult variable to provide access to it, when calling the login event
+     /**
+@@ -65,8 +68,17 @@ public class LoginEvent extends AsyncEvent<LoginEvent> implements Cancellable
+     @Deprecated
+     public String getCancelReason()
+     {
+-        return BaseComponent.toLegacyText( getCancelReasonComponents() );
++        // Waterfall start - we're going on an adventure!
++//        return BaseComponent.toLegacyText( getCancelReasonComponents() );
++        return net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer.legacySection().serialize(cancelReason);
++        // Waterfall end
++    }
++
++    // Waterfall start - add previously Lombok-generated getter - we're going on an adventure!
++    public BaseComponent[] getCancelReasonComponents() {
```

Comment:
Let's go the extra mile and add nullability annotations here too. They don't change ABI, so might as well be nice when we can.